### PR TITLE
fix: apply data_source to CostMapWIdget api and config

### PR DESCRIPTION
**Pull request by WANZARGEN**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [x] Bug fixes
- [x] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description

Please review in order by commits!!!

Main feature
- Added `data-source-reference-store`
- Updated reference handler which is prop for `PJsonSchemaForm`. It is used at widget options edit modal.

### CODE REVIEW

File: apps/web/src/services/dashboards/shared/CustomDateRangeModal.vue [hunk 1]
```diff
@@ -77,10 +77,6 @@ const state = reactive({
     settingsByGranularity: computed<CustomRangeModalSettings>(() => {
         if (!props.granularity) return {};
         const customRangeModalSettingsByGranularity = {
-            [GRANULARITY.ACCUMULATED]: {
-                helpTextFrom: i18n.t('BILLING.COST_MANAGEMENT.COST_ANALYSIS.HELP_TEXT.UP_TO_LAST_12_MONTHS'),
-                dateType: 'yearToMonth',
-            },
```
Comment (yuda110) [inline apps/web/src/services/dashboards/shared/CustomDateRangeModal.vue:83]: Oh, ACCUMULATED removed!

File: apps/web/src/services/cost-explorer/store/data-source-reference-store.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    currency?: Currency;
+}
+type DataSourceItems = Required<Pick<ReferenceItem<DataSourceMetadata>, 'key'|'label'|'name'|'data'>>;
+type DataSourceMap = ReferenceMap<DataSourceItems>;
+const LOAD_TTL = 1000 * 60 * 60 * 3; // 3 hours
+const lastLoadedTimes: Record<string, number> = {
+    dataSource: 0,
+};
+
+export const useDataSourceReferenceStore = defineStore('cost-data-source-reference-store', {
```
Comment (piggggggggy) [inline apps/web/src/services/cost-explorer/store/data-source-reference-store.ts:21]: Oh.. you already made it... what are we going to do..?

**Review by piggggggggy — APPROVED** [review]
LGTM!

File: apps/web/src/services/dashboards/widgets/cost-widgets/cost-map/widget-config.ts [hunk 3]
```diff
@@ -27,12 +27,12 @@ const costMapWidgetConfig: WidgetConfig = {
         granularity: GRANULARITY.YEARLY,
     },
     options_schema: {
-        default_properties: ['cost_group_by', ...getWidgetFilterSchemaPropertyNames('provider', 'project', 'service_account', 'region', 'cost_product', 'cost_account')],
-        fixed_properties: ['cost_group_by'],
+        default_properties: ['data_source', 'cost_group_by', ...getWidgetFilterSchemaPropertyNames('provider', 'project', 'service_account', 'region', 'cost_product', 'cost_account')],
+        fixed_properties: ['data_source', 'cost_group_by'],
```
Comment (yuda110) [inline apps/web/src/services/dashboards/widgets/cost-widgets/cost-map/widget-config.ts:31]: praise: 👍

**Review by yuda110 — APPROVED** [review]
LGTM!!

**Pull request merged**